### PR TITLE
chore(api-reference): speed up config tests

### PR DESCRIPTION
**Pull request by amritk**
## Problem

The api reference config tests were super slow because each test mounts the api reference component which is really heavy.

## Solution

Consolidated the tests into 3 cases while maintaining the same coverage, speeding it up by quite a bit

## Checklist

- [x] I explained why the change is needed.
- [ ] I added a changeset. 
- [x] I added tests.
- [ ] I updated the documentation.

---

> [!NOTE]
> **Low Risk**
> Primarily refactors tests and exposes an additional computed value for testing; no runtime behavior changes beyond removing a debug-only global in the API client modal.
> 
> **Overview**
> Speeds up `ApiReference` configuration test suite by consolidating many per-option mount tests into a few broader cases that assert defaults, common booleans/simple options, callbacks/complex options, and selected edge cases.
> 
> To support the new assertions, `ApiReference.vue` now exposes `sidebarItems` via `defineExpose`, and a debug-only `window.dataDumpWorkspace` hook is removed from the API client `Modal.vue` (the store debug hook remains in `ApiReference.vue`).
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 0ece1afcb7dd79279d3a43dcdaf084eb9c632bf2. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: packages/api-reference/src/components/ApiReference.config.test.ts [hunk 1]
```diff
@@ -1215,28 +463,94 @@
...
+      }
+    }
 
-    await flushPromises()
+    // redirect: function
+    expect(replaceStateSpy).toHaveBeenCalledWith({}, '', '#new')
+
+    // servers: array
+    const ServerSelectorComponent = wrapper.findComponent({ name: 'ServerSelector' })
+    expect(ServerSelectorComponent.text().includes('api.example.com')).toBe(true)
```
[the remark this answers is not in the document]
Comment (amritk) [reply packages/api-reference/src/components/ApiReference.config.test.ts:511]: learn from cursor copilot

**Comment by hanspagel** [comment]
tests seem to fail :D
